cocos/renderer/backend/metal/TextureMTL.mm

```diff
@@ -317,6 +347,12 @@ bool convertData(uint8_t* src, unsigned int length, TextureFormat format, uint8_
 void TextureCubeMTL::updateSamplerDescriptor(const SamplerDescriptor &sampler)
 {
     createSampler(_mtlDevice, sampler);
+    bool needGenerateMipmap = !_isMipmapEnabled && sampler.mipmapEnabled;
+    _isMipmapEnabled = sampler.mipmapEnabled;
+    if(needGenerateMipmap)
+    {
+        generateMipmaps();
+    }
```

Comment:
It seems the logic is duplicated in all Textures(TextureGL, TextureMTL, TextureCubeGL, TextureCubeMTL), can put the codes in Texture.